tests/integrations/test_messagingqueues.py

```diff
@@ -0,0 +1,90 @@
+        self.r = weblog.get("/dsm?integration=kafka")
+
+    def setup_dsm_kafka_default_context_propagation(self):
+        self.r = weblog.get("/dsm?integration=kafka")
+
+    def test_dsm_kafka_default_context_propagation(self):
+        for data in interfaces.agent.get_dsm_data():
+            for stats_bucket in data["request"]["content"].get("Stats", {}):
+
+                producer_stats_point = None
+                consumer_stats_point = None
+
+                for stats_point in stats_bucket.get("Stats", {}):
+                    if stats_point["EdgeTags"][0] == "direction:in":
+                        consumer_stats_point = stats_point
+                    elif stats_point["EdgeTags"][0] == "direction:out":
+                        producer_stats_point = stats_point
+
+                # consumers are a direct child of a producer span
+                assert consumer_stats_point["ParentHash"] == producer_stats_point["Hash"]
+
+    # kafka in dd-trace-py does not propagate span context for tracing purposes
+    def test_apm_kafka_default_context_propagation(self):
+        spans_payload = interfaces.library.get_spans(self.r)
+
+        # get_spans returns several tuple items
+        for items in spans_payload:
+            # each iteration of spans_payload is a tuple
+            data, trace, root_span = items
+            producer_span = None
+            consumer_span = None
+
+            for span in trace:
+                span_component = span["meta"]["component"]
+                if "span.kind" in span["meta"].keys():
+                    span_kind = span["meta"]["span.kind"]
+                else:
+                    span_kind = "none"
+
+                if span_component == "kafka" and span_kind == "producer":
+                    logger.debug("producer span found:", span)
+                    producer_span = span
+                elif span_component == "kafka" and span_kind == "consumer":
+                    logger.debug("consumer span found:", span)
+                    consumer_span = span
+
+            assert producer_span is not None
+            assert consumer_span is not None
+
+            # consumer and producer spans have no direct parent/child relationship
+            assert consumer_span["parent_id"] != producer_span["span_id"]
+            assert producer_span["parent_id"] != consumer_span["span_id"]
+
+            # Check for indirect relationships
+            # Assert that the producer span is never a parent:
+            child_span_of_producer_span = MQHelper.find_span_by_field(producer_span, trace, "span_id", "parent_id")
+            assert child_span_of_producer_span == None
+
+            # Check if the consumer span's ancestors contain the producer span at any point in the segment
+            parent_span = ""
+            starting_span = consumer_span
+            while parent_span != None:
+                parent_span = MQHelper.find_span_by_field(starting_span, trace, "parent_id", "span_id")
```

Comment:
is it safe to assume we can skip this?  (i.e. if we check the children of a producer, that there's no way to have a consumer without a parent producer?)

Reply:
I'm removing this for now, the unknown for me is whether it's possible for a `kafka.consume` span to be orphaned from its parent?

If that's a valid thing to check for and make sure it can't happen, I can bring this idea back and test that `kafka.consume` spans should always be a child of a `kafka.producer` type span.

Reply:
I'll mark this as resolved as https://github.com/DataDog/system-tests/pull/1783 will take over the rest of the apm tracing assertions.